packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -5516,6 +5520,15 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin, Res
   /// Notifies its listeners if the value of [userGestureInProgress] changes.
   final ValueNotifier<bool> userGestureInProgressNotifier = ValueNotifier<bool>(false);
 
+  /// Notifies its listeners if there is a delegate transition from the top route.
+  final ValueNotifier<DelegatedTransitionBuilder?> delegateTransitionBuilderNotifier = ValueNotifier<Widget Function(BuildContext context, Widget? child, Animation<double> animation)?>(null);
+
+  /// Sets the delegate transition.
+  set delegateTransitionBuilder(DelegatedTransitionBuilder? builder) => delegateTransitionBuilderNotifier.value = builder;
```

Comment:
Could we find a way to keep this more private? This is kind of an internal API to the routing machinery that shouldn't be setable for everyone that gets their hands on a NavigatorState via Navigator.of